pubsublite/config.go

```diff
@@ -33,55 +38,52 @@ type TopicConfig struct {
 
 	// The number of partitions in the topic. Must be at least 1. Cannot be
 	// changed after creation.
-	PartitionCount int64
+	PartitionCount int
 
 	// Publish throughput capacity per partition in MiB/s.
 	// Must be >= 4 and <= 16.
-	PublishCapacityMiBPerSec int32
+	PublishCapacityMiBPerSec int
 
 	// Subscribe throughput capacity per partition in MiB/s.
 	// Must be >= 4 and <= 32.
-	SubscribeCapacityMiBPerSec int32
+	SubscribeCapacityMiBPerSec int
 
 	// The provisioned storage, in bytes, per partition. If the number of bytes
 	// stored in any of the topic's partitions grows beyond this value, older
 	// messages will be dropped to make room for newer ones, regardless of the
 	// value of `RetentionDuration`.
 	PerPartitionBytes int64
 
-	// How long a published message is retained. If unset, messages will be
-	// retained as long as the bytes retained for each partition is below
-	// `PerPartitionBytes`.
-	RetentionDuration optional.Duration
+	// How long a published message is retained. If set to `InfiniteRetention`,
+	// messages will be retained as long as the bytes retained for each partition
+	// is below `PerPartitionBytes`.
+	RetentionDuration time.Duration
```

Comment:
Is it valid to have a PerPartitionBytes of zero and RetentionDuration of zero?

Reply:
Nope, good catch. Updated the comments.